examples/gallery/images/grdgradient.py

```diff
@@ -0,0 +1,54 @@
+"""
+Calculating the gradient of a grid file
+--------------------
+The :meth:`pygmt.grgradient` calculate the gradient of a grid file.
+In the example shown below we will see how to calculate an hillshade map based on
+a Data Elevation Model(DEM). :meth:`pygmt.grdgradient` get as input :class:`xarray.DataArray` object
+or a path string to a grid file. We will use the ``radiance`` parameter in order to set the illumination
+source direction and altitude
+"""
+
+import pygmt
+
+fig = pygmt.Figure()
+
+# Define region of interest around Yosemite valley
+region = [-119.825, -119.4, 37.6, 37.825]
+
+# Load sample grid (3 arc second global relief) in target area
+grid = pygmt.datasets.load_earth_relief(resolution="03s", region=region)
+
+# calculate the reflection of a light source projecting from west to east(azimuth 270)
+# and at a latitude of 30 degrees from the horizon
+dgrid = pygmt.grdgradient(grid=grid, radiance=[270, 30])
+
+fig = pygmt.Figure()
+# define figure configuration
+pygmt.config(FORMAT_GEO_MAP="ddd.x", MAP_FRAME_TYPE="plain")
+
+# --------------- plotting the hillshade map -----------
+pygmt.makecpt(cmap="gray", series=[-1.5, 0.3, 0.01])
+fig.grdimage(
+    grid=dgrid,
```

Comment:
Can anyone confirm the definition of a hillshaded map? Based on my own understanding, hillshading means plotting the elevation data `grid`, and also apply the (-1,1) intensity grid `dgrid` for each point. Thus a hillshaded map should be like this (the CPT range should also be changed to reflect the elevation range):
```suggestion
    grid=grid,
    shading=dgrid,
```

Reply:
There is no one answer to this. I think what is plotted in this gallery example could be considered a 'hillshade map' (showing only the hillshade component) but not a 'hillshaded map' implying that the hillshade component alters the appearance of other data (e.g., elevation) that are shown on the map.

My opinion is that it would be better to plot a slope map for the simple grdgradient gallery example and that hillshading should be covered in a different tutorial that can go into different strategies (e.g., multi-directional, mixing slope and hillshade for illumination, transparent overlays versus grdgradient's shading parameter, etc). Sorry for not commenting about this earlier. It will be a great addition to have an example for grdgradient.

Reply:
> 
> 
> There is no one answer to this. I think what is plotted in this gallery example could be considered a 'hillshade map' (showing only the hillshade component) but not a 'hillshaded map' implying that the hillshade component alters the appearance of other data (e.g., elevation) that are shown on the map.
> 
> 
Agree with @meghanrjones regarding this point! I think the example itself is good as is in the current form but would also suggest to add another example/tutorial on this topic.

Reply:
Ill agree that a slops map is more intuitive when you think on the gradient of a DEM. On the other hand, at least in my experience, shade maps are more intuitive to use(you are seeing the topography directly).